lib/couchdb.js

```diff
@@ -0,0 +1,99 @@
+var existsSync = require('fs').existsSync
+var join = require('path').join
+
+var _ = require('lodash')
+var async = require('async')
+var jsonfile = require('jsonfile')
+var log = require('npmlog')
+var randomstring = require('randomstring')
+var request = require('request')
+
+module.exports = function (config, callback) {
+  if (!config.db.url) {
+    var storePath = join(config.paths.data, 'config.json')
+    var storeExists = existsSync(storePath)
+    var store = storeExists && jsonfile.readFileSync(storePath, {
+      throws: false
+    }) || {}
+    var secret = store.couch_httpd_auth_secret
+
+    if (!secret) {
+      secret = randomstring.generate({
+        charset: 'hex'
+      })
+      jsonfile.writeFileSync(storePath, Object.assign(store, {
+        couch_httpd_auth_secret: secret
+      }), {spaces: 2})
+    }
+
+    return callback(null, {
+      secret: secret,
+      authentication_db: '_users'
+    })
+  }
+
+  var couch = request.defaults({
+    baseUrl: config.db.url,
+    json: true
+  })
+
+  async.waterfall([
+    function (callback) {
```

Comment:
That's exactly what I'm about to do, also because that makes it more testable. Just wanted to get this up first :)